cmn/string_test.go

```diff
@@ -0,0 +1,75 @@
+package cmn
+
+import (
+	"math/rand"
+	"testing"
+)
+
+func TestSprintf(t *testing.T) {
+	type args struct {
+		str []string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			args: args{
+				str: []string{"https://", "127.0.0.1", "/resource/test/test.jpg"},
+			},
+			want: "https://127.0.0.1/resource/test/test.jpg",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := Sprintf(tt.args.str...); got != tt.want {
+				t.Errorf("Sprintf() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+func randomString(n int) string {
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letterBytes[rand.Intn(len(letterBytes))]
+	}
+	return string(b)
+}
+
+func plusString(n int, str string) string {
+	s := ""
+	for i := 0; i < n; i++ {
+		s += str
+	}
+	return s
+}
+
+// goos: darwin
+// goarch: amd64
+// pkg: github.com/NVIDIA/aistore/cmn
+// cpu: Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz
+// BenchmarkSprintf-8   	  916281	      1250 ns/op	     256 B/op	       9 allocs/op
+// PASS
+// ok  	github.com/NVIDIA/aistore/cmn	2.679s
+func BenchmarkSprintf(b *testing.B) {
+	for n := 0; n < b.N; n++ {
+		Sprintf(randomString(10), randomString(20), randomString(30))
+	}
+}
+
+// goos: darwin
+// goarch: amd64
+// pkg: github.com/NVIDIA/aistore/cmn
+// cpu: Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz
+// BenchmarkPlus-8   	       7	 147251687 ns/op	1537847900 B/op	   10050 allocs/op
+// PASS
+// ok  	github.com/NVIDIA/aistore/cmn	2.373s
+func BenchmarkPlus(b *testing.B) {
+	for n := 0; n < b.N; n++ {
+		plusString(10000, randomString(30))
```

Comment:
Same comment here about prepartions. And also this benchmark is hard to compare as you do plus `10000` times. Which you don't in `Sprintf`. Could you make those 2 benchmarks comparable?

I would suggest doing something like:
```
b.Run(func(b *B) {
	for n := 0; n < b.N; n++ {
 		Sprintf(preparedStrings)
 	}
}
b.Run(func(b *B) {
	for n := 0; n < b.N; n++ {
 		plusString(preparedStrings)
 	}
}
```
then you are using the same string and then you can conclude something with that.